src/platform_impl/windows/dark_mode.rs

```diff
@@ -0,0 +1,257 @@
+            assert_eq!(
+                0,
+                uxtheme::SetWindowTheme(hwnd, theme_name as _, std::ptr::null())
+            );
+
+            set_dark_mode_for_window(hwnd, is_dark_mode)
+        }
+
+        is_dark_mode
+    } else {
+        false
+    }
+}
+
+fn set_dark_mode_for_window(hwnd: HWND, is_dark_mode: bool) {
+    // Uses Windows undocumented API SetWindowCompositionAttribute,
+    // as seen in win32-darkmode example linked at top of file.
+
+    type SetWindowCompositionAttribute =
+        unsafe extern "system" fn(HWND, *mut WINDOWCOMPOSITIONATTRIBDATA) -> BOOL;
+
+    #[allow(non_snake_case)]
+    type WINDOWCOMPOSITIONATTRIB = u32;
+    const WCA_USEDARKMODECOLORS: WINDOWCOMPOSITIONATTRIB = 26;
+
+    #[allow(non_snake_case)]
+    #[repr(C)]
+    struct WINDOWCOMPOSITIONATTRIBDATA
+    {
+        Attrib: WINDOWCOMPOSITIONATTRIB,
+        pvData: PVOID,
+        cbData: SIZE_T
+    }
+
+    lazy_static! {
+        static ref SET_WINDOW_COMPOSITION_ATTRIBUTE:
+        SetWindowCompositionAttribute = {
+            unsafe {
+                let module = libloaderapi::LoadLibraryExW(
+                    widestring("user32.dll").as_ptr(),
+                    std::ptr::null_mut(),
+                    libloaderapi::LOAD_LIBRARY_SEARCH_SYSTEM32,
+                );
+
+                let handle = libloaderapi::GetProcAddress(
+                    module,
+                    "SetWindowCompositionAttribute\0".as_ptr() as _,
+                );
+
+                if handle.is_null() {
+                    unsafe extern "system" fn always_false(
+                        _: HWND,
+                        _: *mut WINDOWCOMPOSITIONATTRIBDATA
+                    ) -> BOOL {
+                        false.into()
+                    }
+                    always_false
+                } else {
+                    std::mem::transmute(handle)
+                }
+            }
+        };
+    }
+
+    unsafe {
+        // SetWindowCompositionAttribute needs a bigbool (i32), not bool.
+        let mut is_dark_mode_bigbool = is_dark_mode as BOOL;
+
+        let mut data = WINDOWCOMPOSITIONATTRIBDATA {
+            Attrib: WCA_USEDARKMODECOLORS,
+            pvData: &mut is_dark_mode_bigbool as *mut _ as _,
+            cbData: std::mem::size_of_val(&is_dark_mode_bigbool) as _
+        };
+
+        assert_eq!(
+            1,
+            SET_WINDOW_COMPOSITION_ATTRIBUTE(
+                hwnd,
+                &mut data as *mut _
+            )
+        );
+    }
+}
+
+fn should_use_dark_mode() -> bool {
+    should_apps_use_dark_mode() && !is_high_contrast()
+}
+
+fn should_apps_use_dark_mode() -> bool {
+    type ShouldAppsUseDarkMode = unsafe extern "system" fn() -> bool;
+    lazy_static! {
+        static ref SHOULD_APPS_USE_DARK_MODE: ShouldAppsUseDarkMode = {
+            unsafe {
+
+                const UXTHEME_SHOULDAPPSUSEDARKMODE_ORDINAL: WORD = 132;
+
+                let module = libloaderapi::LoadLibraryExW(
```

Comment:
We are using `LoadLibraryA` in the rest of the library, not sure if this is important to anybody to keep this uniform.